core/src/main/java/com/cloud/storage/template/HttpTemplateDownloader.java

```diff
@@ -219,7 +219,7 @@ public long download(boolean resume, DownloadCompleteCallback callback) {
         } catch (IOException ioe) {
             status = TemplateDownloader.Status.UNRECOVERABLE_ERROR; //probably a file write error?
             // Let's not overwrite the original error message.
-            if (errorString.isEmpty()){
+            if (errorString.equals(" ")){
```

Comment:
i'd go for `errorString.trim().isEmpty()` to be honest (to be more defensive, but this should work as well. And it might be wise to alter the initialisation statement as well???

Reply:
What if we get rid of the check altogether and append the error message to what was there already?